### ISSUE TITLE
m.as_signal progressbar always shows

### ISSUE THREAD

**Issue by thomasaarholt** [hyperspy/hyperspy#1758]
On eels spectra, s.remove_background(show_progressbar=False) still shows the progressbar.

**Comment by thomasaarholt** [comment]
The error stems from m.as_signal() producing a progressbar.

**Comment by thomasaarholt** [comment]
I've found the error, but my lambda python isn't good enough to understand what's happening.
It looks like show_progressbar just has the wrong argument entirely?
model.py line 499:
```python
with ThreadPoolExecutor(max_workers=parallel) as exe:
                _map = exe.map(
                    lambda thing: thing[0]._as_signal_iter(
                        data=thing[1],
                        component_list=component_list,
                        out_of_range_to_nan=out_of_range_to_nan,
                        show_progressbar=thing[2] + 1),
                    zip(models, data_slices, range(int(parallel))))
            _ = next(_map)
```

Edit: oh, it's to track the progressbar. Fix in #1759

**Comment by francisco-dlp** [comment]
#1758
